### DISCUSSION TITLE
Something I'd be happy if users would read before asking for features

### DISCUSSION THREAD

**Discussion by icflorescu** [icflorescu/mantine-datatable#525] [Announcements]
There are always users keenly asking for features that would satisfy their particular use-cases, but without really taking into consideration how those features would impact other pre-existing features, or the overall performance of the library. Sorting through such feature requests and deciding which ones make sense to implement is something that should be done with extreme care, in my experience.

There are also users that raise feature requests motivated by the idea that - and I quote - _"Mantine React Table has it"_. Which, in my opinion, is not a good enough reason. Different packages have different - and often contradictory - feature sets and approaches, and users are free to choose between them.

For instance, here were a few key points that were very important for me when I started this library:

- Lightweight (and we're still around 10k gzipped and compressed, I think)
- Great UX for scrolling - hence the shadow visual cues for both vertical and horizontal scrolling
- [Gmail-style batch record selection with shift-click](https://icflorescu.github.io/mantine-datatable/examples/records-selection/) - something that tanstack tables don't do, AFAIK

So, if you're planning to ask for a feature simply because _"this other library has it"_, please consider pairing it with an implementation PR, or using that other library.

As I previously mentioned with a few other occasions, Mantine DataTable is my dearest open-source project, but unfortunately it doesn't make money for me and I am therefore not in the position to dedicate as much time as I (or you) would like for it.
If your company is benefiting from using the library in its projects, you could change that by coming up with a [recurring sponsorship](https://github.com/sponsors/icflorescu).

**Comment by KevinVandy** [comment]
All of the features in MRT have a price in bundle size. Also upgrading every feature perfectly to the new version of Mantine takes even more time. I think MDT has chosen a very good set of features where I point a lot of people who want to use a lighter weight version of MRT here to your project. Great job on your Mantine v7 upgrade, as I'm still months into my own Mantine v7 upgrade for MRT v2.

**Reply by icflorescu** [reply]
As always, you're very kind, @KevinVandy!
Thanks for the nice words and for your sponsorship.
Kudos for building Mantine React Table. It's definitely something I'm pointing people to when they look for features MDT doesn't have.
